src/shared/rm-rf.c

```diff
@@ -44,7 +45,7 @@ static int patch_dirfd_mode(
         if (st.st_uid != geteuid())  /* this only works if the UID matches ours */
                 return -EACCES;
 
-        if (fchmod(dfd, (st.st_mode | 0700) & 07777) < 0)
+        if (fchmod_opath(dfd, (st.st_mode | 0700) & 07777) < 0)
```

Comment:
```c
r = chmod_opath(…)
if (r < 0)
        return r;
```

i.e. in the failure path fchmod_opath() does an proc_mounted() which might alter errno. Hence it's essential we actually look at the return value, not errno

Reply:
Ouch.